### ISSUE TITLE
Remove lib prefix

### ISSUE THREAD

**Issue by Flamefire** [ComputationalRadiationPhysics/halt#15]
As @psychocoderHPC just mentioned it is better to drop the "lib" prefix from the name as otherwise a precompiled library would be called "liblibLiFFT" which is kinda redundant.
As the namespace is already only "lifft" one could keep the general name.

This would be the same as libSplash: https://github.com/ComputationalRadiationPhysics/libSplash The full name is libSplash but the namespace and folders are only "splash"

CCing @ax3l @bussmann

**Comment by ax3l** [comment]
Sure!

**Comment by ax3l** [comment]
yes the idea is to call the repo itself something like `liFFT` or `lifft` for easy directory naming and to use the official spelling `libLiFFT` to make obvious it is a library.

namespaces, CMake project name / targets, coding prefixes, etc. should simply be named without the `lib` as you suggested, `lifft` or `liFFT` as you prefer. Actual library archives (`.so`, `.a`, ...) are then automatically prefixed with `lib` and linking can be done via `-Lpath -llifft` which selects `liblifft` files.
